secondarylib/src/main/scala/com/socrata/datacoordinator/secondary/SecondaryWatcher.scala

```diff
@@ -207,16 +288,53 @@ class SecondaryWatcherClaimManager(dsInfo: DSInfo, claimantId: UUID, claimTimeou
     // concurrency.  We could potentially have a separate pool for the claim manager.
     using(dsInfo.dataSource.getConnection()) { conn =>
       using(conn.prepareStatement(
-        """UPDATE secondary_manifest
-        |SET claimed_at = CURRENT_TIMESTAMP
-        |WHERE claimant_id = ?""".stripMargin)) { stmt =>
+        s"""UPDATE secondary_manifest
+          |SET claimed_at = CURRENT_TIMESTAMP
+          |WHERE claimant_id = ?${SecondaryWatcherClaimManager.andInWorkingSetSQL}""".stripMargin)) { stmt =>
         stmt.setObject(1, claimantId)
         stmt.executeUpdate()
       }
     }
   }
 }
 
+object SecondaryWatcherScheduledExecutor {
+  val log = LoggerFactory.getLogger(classOf[SecondaryWatcher[_,_]])
+
+  private val scheduler: ScheduledExecutorService = Executors.newScheduledThreadPool(1)
+
+  def shutdown() = SecondaryWatcherScheduledExecutor.scheduler.shutdownNow()
+
+  def schedule(shouldExit: => Boolean, name: String, interval: FiniteDuration): Unit = {
+    val runnable = new Runnable() {
+      def run(): Unit = {
+        try {
+          Thread.currentThread().setName("SecondaryWatcher scheduled health checker")
+          log.debug("Running scheduled health check of: {}", name)
+          if (shouldExit) {
+            log.error("Failed scheduled health check of: {}; exiting.", name)
```

Comment:
Yeah, it is, maybe I should rename `shouldExit` to something more failure-ish sounding